flytekit/extras/persistence/http.py

```diff
@@ -0,0 +1,68 @@
+import os
+import pathlib
+
+import requests as _requests
+
+from flytekit.common.exceptions import user as _user_exceptions
+from flytekit.core.data_persistence import DataPersistence, DataPersistencePlugins
+from flytekit.loggers import logger
+
+
+class HttpPersistence(DataPersistence):
+    """
+    DataPersistence implementation for the HTTP protocol. only supports downloading from an http source. Uploads are
+    not supported currently.
+    """
+
+    PROTOCOL_HTTP = "http"
+    PROTOCOL_HTTPS = "https"
+    _HTTP_OK = 200
+    _HTTP_FORBIDDEN = 403
+    _HTTP_NOT_FOUND = 404
+
+    def __init__(self, *args, **kwargs):
+        super(HttpPersistence, self).__init__(name="http/https", *args, **kwargs)
+
+    def exists(self, path: str):
+        rsp = _requests.head(path)
+        allowed_codes = {
+            type(self)._HTTP_OK,
+            type(self)._HTTP_NOT_FOUND,
+            type(self)._HTTP_FORBIDDEN,
```

Comment:
is the `type` call necessary? (see below as well)

Reply:
nope, this is legacy code, removing - good call